script.sh

```diff
@@ -25,70 +25,75 @@ echo '::group::Preparing'
+  fi
+
   if [[ -z "${TFLINT_PLUGIN_DIR}" ]]; then
     export TFLINT_PLUGIN_DIR="${TFLINT_PATH}/.tflint.d/plugins"
     mkdir -p "${TFLINT_PLUGIN_DIR}"
-    echo "TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}" >> "$GITHUB_ENV"
+    echo "TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}" >> "${GITHUB_ENV}"
   else
     echo "Found pre-configured TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}"
   fi
 echo '::endgroup::'
 
-echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
+echo "::group::🐶 Installing reviewdog (${REVIEWDOG_VERSION}) ... https://github.com/reviewdog/reviewdog"
   curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${REVIEWDOG_PATH}" "${REVIEWDOG_VERSION}" 2>&1
 
-  echo "$REVIEWDOG_PATH" >> "$GITHUB_PATH"
-  export PATH="$REVIEWDOG_PATH:$PATH"
+  echo "${REVIEWDOG_PATH}" >> "${GITHUB_PATH}"
+  export PATH="${REVIEWDOG_PATH}:${PATH}"
 echo '::endgroup::'
 
-echo '::group:: Installing tflint ... https://github.com/terraform-linters/tflint'
-  if [[ -z "${INPUT_TFLINT_VERSION}" ]] || [[ "${INPUT_TFLINT_VERSION}" == "latest" ]]; then
-    echo "Looking up the latest version ..."
-    version=$(curl --silent "https://api.github.com/repos/terraform-linters/tflint/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
-  else
-    version=${INPUT_TFLINT_VERSION}
-  fi
-
+echo "::group:: Installing tflint (${tflint_version}) ... https://github.com/terraform-linters/tflint"
   curl --silent --show-error --fail \
-    --location "https://github.com/terraform-linters/tflint/releases/download/${version}/tflint_${os}_${arch}.zip" \
+    --location "https://github.com/terraform-linters/tflint/releases/download/${tflint_version}/tflint_${os}_${arch}.zip" \
     --output "${TEMP_PATH}/tflint.zip"
 
   unzip -u "${TEMP_PATH}/tflint.zip" -d "${TEMP_PATH}/temp-tflint"
   test ! -d "${TFLINT_PATH}" && install -d "${TFLINT_PATH}"
   install "${TEMP_PATH}/temp-tflint/tflint" "${TFLINT_PATH}"
   rm -rf "${TEMP_PATH}/tflint.zip" "${TEMP_PATH}/temp-tflint"
 
-  echo "${TFLINT_PATH}" >> "$GITHUB_PATH"
-  export PATH="${TFLINT_PATH}:$PATH"
+  echo "${TFLINT_PATH}" >> "${GITHUB_PATH}"
+  export PATH="${TFLINT_PATH}:${PATH}"
 echo '::endgroup::'
 
-
 for RULESET in ${INPUT_TFLINT_RULESETS}; do
   PLUGIN="tflint-ruleset-${RULESET}"
   REPOSITORY="https://github.com/terraform-linters/${PLUGIN}"
 
-  echo "::group:: Installing tflint plugin for ${RULESET} ... ${REPOSITORY}"
+  echo "::group:: Installing tflint plugin for ${RULESET} (latest) ... ${REPOSITORY}"
     curl --silent --show-error --fail \
       --location "${REPOSITORY}"/releases/latest/download/"${PLUGIN}"_"${os}"_"${arch}".zip \
       --output "${PLUGIN}".zip \
     && unzip "${PLUGIN}".zip -d "${TFLINT_PLUGIN_DIR}" && rm "${PLUGIN}".zip
   echo '::endgroup::'
 done
 
-echo "::group:: Print tflint version details"
+echo "::group:: Print tflint details ..."
   tflint --version
 echo '::endgroup::'
 
 
+
```

Comment:
?

Reply:
The script got rather long, so I added an extra line to separate the big logical chunks: the preparation (dependencies and all that) from actually running `reviewdog`. I think it looks a bit clearer like that.

Reply:
One blank line I get -- splitting paragraphs
Two blank lines I get -- splitting sections

Three blank lines?

I think that here (seen in a different view w/ more context), you should be removing this blank line because it was associated with:
```
export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
```

which you also removed.